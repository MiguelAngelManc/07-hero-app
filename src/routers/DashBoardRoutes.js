import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom/cjs/react-router-dom.min'
import { DcScreen } from '../Components/dc/DcScreen'
import { HeroScreen } from '../Components/heroes/HeroScreen'
import { MarvelScreen } from '../Components/marvel/MarvelScreen'
import { SearchScreen } from '../Components/search/SearchScreen'
import { Navbar } from '../Components/ui/Navbar'

export const DashBoardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container">

                <Switch>

                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroeId" component={ HeroScreen }/>
                    <Route exact path="/dc" component={ DcScreen }/>
                    <Route path="/search" component={ SearchScreen }/>

                    <Redirect to="/marvel"/>
                    
                </Switch>

            </div>
        </>
    )
}
