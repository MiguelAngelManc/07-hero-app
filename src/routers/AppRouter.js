import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link
} from "react-router-dom";
import { DcScreen } from "../Components/dc/DcScreen";
import { LoginScreen } from "../Components/login/LoginScreen";
import { MarvelScreen } from "../Components/marvel/MarvelScreen";
import { Navbar } from "../Components/ui/Navbar";
import { DashBoardRoutes } from "./DashBoardRoutes";

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <Switch>
                    <Route exact path="/login" component={ LoginScreen } />
                    
                    <Route path="/" component={ DashBoardRoutes } />
                    

                </Switch>
            </div>
        </Router>
    );
}
    
