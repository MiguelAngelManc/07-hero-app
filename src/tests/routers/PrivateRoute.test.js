
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom"
import { PrivateRoute } from "../../routers/PrivateRoute"


describe('Pruebas en PrivateRoute', () => {
    
    const props = {
        location:{
            pathname:"testpruebas"
        }
    }

    Storage.prototype.setItem = jest.fn();

    test('Muestra el compoentne si esta autenticado y guarda en localsotrage', () => {
        
        const wrapper = mount(
        <MemoryRouter>
            <PrivateRoute 
                isAuth={true}
                component={ () => <span>hola</span>}
                {...props}
            />
        </MemoryRouter>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('span').exists()).toBe(true)
        expect(localStorage.setItem).toHaveBeenCalledWith('lastpath','testpruebas')
        
    })

    test('Probar el componente si no esta autenticado',()=>{

        const wrapper = mount(
            <MemoryRouter>
                <PrivateRoute 
                    isAuth={false}
                    component={ () => <span>hola</span>}
                    {...props}
                />
            </MemoryRouter>);

        expect(wrapper.html()).toBe('')

    })
    

})
