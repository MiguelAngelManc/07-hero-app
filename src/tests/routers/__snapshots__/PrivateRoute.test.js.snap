// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrivateRoute Muestra el compoentne si esta autenticado y guarda en localsotrage 1`] = `
<span>
  hola
</span>
`;
